apps/camera/js/lib/camera/camera.js

```diff
@@ -981,77 +1000,60 @@ Camera.prototype.stoppedRecording = function(recorded) {
   this.orientation.start();
 
   var self = this;
-  var elapsedTime = self.get('videoElapsed');
-  var filepath;
-  var req;
-  var storage;
+  var videoReq;
+  var posterReq;
+  var video;
+  var got = 0;
+
   if (recorded) {
-    // Re-fetch the blob from storage
-    storage = this.storage.video;
-    filepath = this.video.filepath;
-    req = storage.get(filepath);
-    req.onerror = onError;
-    req.onsuccess = onSuccess;
+    video = {
+      filepath: this.video.filepath,
+      rotation: this.video.rotation,
+      width: this.video.width,
+      height: this.video.height,
+      poster: {
+        filepath: this.video.poster.filepath,
+        rotation: this.video.poster.rotation,
+        width: this.video.poster.width,
+        height: this.video.poster.height
+      }
+    };
```

Comment:
Good point. Yeah, if you're just doing a deep copy of this object, you can do the following:

```
var mix = require('lib/mixin');
video = mix({}, this.video);
```

Basically, the `mix()` function is for copying properties from one object to another. But, if you specify a new empty object `{}` for the first parameter, it acts as a "copy" function.